src/main/java/com/google/sps/servlets/PotentialMatchesServlet.java

```diff
@@ -0,0 +1,109 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import java.util.List;
+import java.util.ArrayList;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.sps.data.friend_map.UserNode;
+import com.google.sps.data.friend_map.UserFriendsMap;
+import com.google.sps.data.potentialMatchAlgorithm;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.Entity;
+import com.google.common.collect.ImmutableSet;
+
+/**
+* Handles requests for getting the next potential match for a user's feed page.
+ */
+@WebServlet("/potential-matches")
+public class PotentialMatchesServlet extends HttpServlet {
+  private static final String USER_ENTITY = "User";
+  private static final String USER_ID_PROPERTY = "id";
+  private static final String USER_FRIENDS_LIST_PROPERTY = "friends-list";
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String currUserID = request.getParameter("userid");
+    
+    String nextPotentialMatchID = loadUserPotentialMatch(currUserID);
+
+    response.setContentType("text/html");
+    response.getWriter().print(nextPotentialMatchID);
+  }
+
+  /**
+  * Loads the next potential match for a user
+  *
+  * <p>Currently is just loading all the possible matches and obtaining one of them.
+  *
+  * <p>TODO(#19): Obtain potential match results from datastore instead of running the 
+  * potential match finding methods repeatedly.
+  *
+  * @param userID The userID of the user who's potential match is being found
+  * @return The next potential match for a user
+  */
+  private String loadUserPotentialMatch(String userID) {
+    ImmutableSet<UserNode> userNodes = createUserNodes();
+    UserFriendsMap friendsMap = new UserFriendsMap(userNodes);
+    
+    ImmutableSet<String> potentialMatches = potentialMatchAlgorithm.findPotentialMatchesForUser(userID, friendsMap);
+  
+    String nextPotentialMatchID = potentialMatches.size() > 0 ?
+      potentialMatches.iterator().next() : "NO_POTENTIAL_MATCHES";
+    
+    return nextPotentialMatchID;
+  }
+
+  /**
+  * Helper method that creates the set of UserNodes that will be fed into the potential
+  * matching method
+  *
+  * @return the set of user nodes for all current app users
+  */
+  private ImmutableSet<UserNode> createUserNodes() {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(new Query(USER_ENTITY));
```

Comment:
Possibly: `DatastoreServiceFactory.getDatastoreService().prepare(new Query(USER_ENTITY));`
However, the `kind` for user entities, at least for the user entities that I'm storing, is their user id.

Reply:
So each `kind` corresponds to a single user entity.

Reply:
Discussed and resolved offline